src/material-experimental/mat-feature-highlight/feature-highlight-content-directives.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Input, OnInit, Optional} from '@angular/core';
+
+import {FeatureHighlight} from './feature-highlight';
+import {FeatureHighlightRef} from './feature-highlight-ref';
+
+/**
+ * Target featured element to be highlighted. Use it to style the z-index and
+ * position correctly so that the target element is displayed on top of other
+ * feature highlight elements.
+ */
+@Directive({
+  selector: `[feature-highlight-target],
+       [featureHighlightTarget],
+       feature-highlight-target`,
+  exportAs: 'featureHighlightTarget',
+  host: {
+    '[style.zIndex]': 'zIndex',
+    '[style.position]': 'position',
+  },
+})
+export class FeatureHighlightTarget {
+  @Input() featureHighlightEnabled = true;
+
+  /**
+   * Return the zIndex value for the target element. When outer circle is not
+   * bounded, zIndex of the target element needs to be 1001 as the overlay
+   * container has zIndex of 1000. When outer circle is bounded, zIndex only
+   * needs to be 3.
+   */
+  get zIndex(): string|null {
+    return this.featureHighlightEnabled ? '1001' : null;
+  }
```

Comment:
We prefer using methods rather than getters for anything that doesn't need to be a bindable property since the getter generates a bit more es5 code.